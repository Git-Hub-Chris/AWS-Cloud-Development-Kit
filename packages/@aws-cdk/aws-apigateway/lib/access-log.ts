import { ILogGroup } from '@aws-cdk/aws-logs';

/**
 * A API Gateway custom access log destination.
 */
export interface IAccessLogDestination {
  /**
   * Binds destination.
   */
  bind(): AccessLogDestinationConfig
}

/**
 * Options when binding a destination to a API Gateway
 */
export interface AccessLogDestinationConfig {
  /**
   * The Amazon Resource Name (ARN) of the destination resource
   */
  readonly destinationArn: string;
}

/**
 * Use CloudWatch Logs as a custom access log destination for API Gateway.
 */
export class LogGroupLogDestination implements IAccessLogDestination {
  constructor(private readonly logGroup: ILogGroup) {
  }

  /**
   * Binds this destination to the CloudWatch Logs.
   */
  public bind(): AccessLogDestinationConfig {
    return {
      destinationArn: this.logGroup.logGroupArn
    };
  }
}

/**
 * $context variables when customize access log.
 */
export class AccessLogField {
  /**
   * The API owner's AWS account ID.
   */
  public static contextAccountId() {
    return '$context.requestId';
  }

  /**
   * The identifier API Gateway assigns to your API.
   */
  public static contextApiId() {
    return '$context.apiId';
  }

  /**
   * A property of the claims returned from the Amazon Cognito user pool after the method caller is successfully authenticated.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-integrate-with-cognito.html
   *
   * @param property A property key of the claims.
   */
  public static contextAuthorizerClaims(property: string) {
    return `$context.authorizer.claims.${property}`;
  }

  /**
   * The principal user identification associated with the token sent by the client and returned
   * from an API Gateway Lambda authorizer (formerly known as a custom authorizer).
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html
   */
  public static contextAuthorizerPrincipalId() {
    return '$context.authorizer.principalId';
  }

  /**
   * The stringified value of the specified key-value pair of the `context` map returned from an API Gateway Lambda authorizer function.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-use-lambda-authorizer.html
   * @param property key of the context map.
   */
  public static contextAuthorizer(property: string) {
    return `$context.authorizer.${property}`;
  }

  /**
   * The AWS endpoint's request ID.
   */
  public static contextAwsEndpointRequestId() {
    return '$context.awsEndpointRequestId';
  }

  /**
   * The full domain name used to invoke the API. This should be the same as the incoming `Host` header.
   */
  public static contextDomainName() {
    return '$context.domainName';
  }

  /**
   * The first label of the `$context.domainName`. This is often used as a caller/customer identifier.
   */
  public static contextDomainPrefix() {
    return '$context.domainPrefix';
  }

  /**
   * A string containing an API Gateway error message.
   * This variable can only be used for simple variable substitution in a GatewayResponse body-mapping template,
   * which is not processed by the Velocity Template Language engine, and in access logging.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-logging.html
   *                       https://docs.aws.amazon.com/apigateway/latest/developerguide/customize-gateway-responses.html
   */
  public static contextErrorMessage() {
    return '$context.error.message';
  }

  /**
   * The quoted value of $context.error.message, namely "$context.error.message".
   */
  public static contextErrorMessageString() {
    return '$context.error.messageString';
  }

  /**
   * A type of GatewayResponse. This variable can only be used for simple variable substitution in a GatewayResponse body-mapping template,
   * which is not processed by the Velocity Template Language engine, and in access logging.
   *
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-logging.html
   *                       https://docs.aws.amazon.com/apigateway/latest/developerguide/customize-gateway-responses.html
   */
  public static contextErrorResponseType() {
    return '$context.error.responseType';
  }

  /**
   * A string containing a detailed validation error message.
   */
  public static contextErrorValidationErrorString() {
    return '$context.error.validationErrorString';
  }

  /**
   * The extended ID that API Gateway assigns to the API request, which contains more useful information for debugging/troubleshooting.
   */
  public static contextExtendedRequestId() {
    return '$context.extendedRequestId';
  }

  /**
   * The HTTP method used. Valid values include: `DELETE`, `GET`, `HEAD`, `OPTIONS`, `PATCH`, `POST`, and `PUT`.
   */
  public static contextHttpMethod() {
    return '$context.httpMethod';
  }

  /**
   * The AWS account ID associated with the request.
   */
  public static contextIdentityAccountId() {
    return '$context.identity.accountId';
  }

  /**
   * For API methods that require an API key, this variable is the API key associated with the method request.
   * For methods that don't require an API key, this variable is
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-usage-plans.html
   */
  public static contextIdentityApiKey() {
    return '$context.identity.apiKey';
  }

  /**
   * The API key ID associated with an API request that requires an API key.
   */
  public static contextIdentityApiKeyId() {
    return '$context.identity.apiKeyId';
  }

  /**
   * The principal identifier of the caller making the request.
   */
  public static contextIdentityCaller() {
    return '$context.identity.caller';
  }

  /**
   * The Amazon Cognito authentication provider used by the caller making the request.
   * Available only if the request was signed with Amazon Cognito credentials.
   * For more information: https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-identity.html
   */
  public static contextIdentityCognitoAuthenticationProvider() {
    return '$context.identity.cognitoAuthenticationProvider';
  }

  /**
   * The Amazon Cognito authentication type of the caller making the request.
   * Available only if the request was signed with Amazon Cognito credentials.
   */
  public static contextIdentityCognitoAuthenticationType() {
    return '$context.identity.cognitoAuthenticationType';
  }

  /**
   * The Amazon Cognito identity ID of the caller making the request. Available only if the request was signed with Amazon Cognito credentials.
   */
  public static contextIdentityCognitoIdentityId() {
    return '$context.identity.cognitoIdentityId';
  }

  /**
   * The Amazon Cognito identity pool ID of the caller making the request.
   * Available only if the request was signed with Amazon Cognito credentials.
   */
  public static contextIdentityCognitoIdentityPoolId() {
    return '$context.identity.cognitoIdentityPoolId';
  }

  /**
   * The AWS organization ID.
   */
  public static contextIdentityPrincipalOrgId() {
    return '$context.identity.principalOrgId';
  }

  /**
   * The source IP address of the TCP connection making the request to API Gateway.
   * Warning: You should not trust this value if there is any chance that the `X-Forwarded-For` header could be forged.
   */
  public static contextIdentitySourceIp() {
    return '$context.identity.sourceIp';
  }

  /**
   * The principal identifier of the user making the request. Used in Lambda authorizers.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-lambda-authorizer-output.html
   */
  public static contextIdentityUser() {
    return '$context.identity.user';
  }

  /**
   * The User-Agent header of the API caller.
   */
  public static contextIdentityUserAgent() {
    return '$context.identity.userAgent';
  }

  /**
   * The Amazon Resource Name (ARN) of the effective user identified after authentication.
   * For more information: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html
   */
  public static contextIdentityUserArn() {
    return '$context.identity.userArn';
  }

  /**
   * The request path.
   * For example, for a non-proxy request URL of https://{rest-api-id.execute-api.{region}.amazonaws.com/{stage}/root/child,
   * the $context.path value is /{stage}/root/child.
   */
  public static contextPath() {
    return '$context.path';
  }

  /**
   * The request protocol, for example, HTTP/1.1.
   */
  public static contextProtocol() {
    return '$context.protocol';
  }

  /**
   * The ID that API Gateway assigns to the API request.
   */
  public static contextRequestId() {
    return '$context.requestId';
  }

  /**
   * The request header override.
   * If this parameter is defined, it contains the headers to be used instead of the HTTP Headers that are defined in the Integration Request pane.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-override-request-response-parameters.html
   *
   * @param headerName
   */
  public static contextRequestOverrideHeader(headerName: string) {
    return `$context.requestOverride.header.${headerName}`;
  }

  /**
   * The request path override. If this parameter is defined,
   * it contains the request path to be used instead of the URL Path Parameters that are defined in the Integration Request pane.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-override-request-response-parameters.html
   *
   * @param pathName
   */
  public static contextRequestOverridePath(pathName: string) {
    return `$context.requestOverride.path.${pathName}`;
  }

  /**
   * The request query string override.
   * If this parameter is defined, it contains the request query strings to be used instead
   * of the URL Query String Parameters that are defined in the Integration Request pane.
   *
   * @param querystringName
   */
  public static contextRequestOverrideQuerystring(querystringName: string) {
    return `$context.requestOverride.querystring.${querystringName}`;
  }

  /**
   * The response header override.
   * If this parameter is defined, it contains the header to be returned instead of the Response header
   * that is defined as the Default mapping in the Integration Response pane.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-override-request-response-parameters.html
   *
   * @param headerName
   */
  public static contextResponseOverrideHeader(headerName: string) {
    return `$context.responseOverride.header.${headerName}`;
  }

  /**
   * The response status code override.
   * If this parameter is defined, it contains the status code to be returned instead of the Method response status
   * that is defined as the Default mapping in the Integration Response pane.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-override-request-response-parameters.html
   */
  public static contextResponseOverrideStatus() {
    return '$context.responseOverride.status';
  }

  /**
   * The CLF-formatted request time (dd/MMM/yyyy:HH:mm:ss +-hhmm).
   */
  public static contextRequestTime() {
    return '$context.requestTime';
  }

  /**
   * The Epoch-formatted request time.
   */
  public static contextRequestTimeEpoch() {
    return '$context.requestTimeEpoch';
  }

  /**
   * The identifier that API Gateway assigns to your resource.
   */
  public static contextResourceId() {
    return '$context.resourceId';
  }

  /**
   * The path to your resource.
   * For example, for the non-proxy request URI of `https://{rest-api-id.execute-api.{region}.amazonaws.com/{stage}/root/child`,
   * The $context.resourcePath value is `/root/child`.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-step-by-step.html
   */
  public static contextResourcePath() {
    return '$context.resourcePath';
  }

  /**
   * The deployment stage of the API request (for example, `Beta` or `Prod`).
   */
  public static contextStage() {
    return '$context.stage';
  }

  /**
   * The response received from AWS WAF: `WAF_ALLOW` or `WAF_BLOCK`. Will not be set if the stage is not associated with a web ACL.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-aws-waf.html
   */
  public static contextWafResponseCode() {
    return '$context.wafResponseCode';
  }

  /**
   * The complete ARN of the web ACL that is used to decide whether to allow or block the request.
   * Will not be set if the stage is not associated with a web ACL.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-aws-waf.html
   */
  public static contextWebaclArn() {
    return '$context.webaclArn';
  }

  /**
   * The trace ID for the X-Ray trace.
   * For more information: https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-enabling-xray.html
   */
  public static contextXrayTraceId() {
    return '$context.xrayTraceId';
  }

  /**
   * The authorizer latency in ms.
   */
  public static contextAuthorizerIntegrationLatency() {
    return '$context.authorizer.integrationLatency';
  }

  /**
   * The integration latency in ms.
   */
  public static contextIntegrationLatency() {
    return '$context.integrationLatency';
  }

  /**
   * For Lambda proxy integration, this parameter represents the status code returned from AWS Lambda,
   * not from the backend Lambda function.
   */
  public static contextIntegrationStatus() {
    return '$context.integrationStatus.';
  }

  /**
   * The response latency in ms.
   */
  public static contextResponseLatency() {
    return '$context.responseLatency';
  }

  /**
   * The response payload length.
   */
  public static contextResponseLength() {
    return '$context.responseLength';
  }

  /**
   * The method response status.
   */
  public static contextStatus() {
    return '$context.status';
  }
}

/**
 * A API Gateway custom access log format.
 */
export interface AccessLogOptions {
  /**
   * A API Gateway custom access log format
   */
  readonly format: string;
}

/**
 * jsonWithStandardFields property
 */
export interface IJsonWithStandardFieldProps {
  /**
   * If this flag is enabled, $context.identity.sourceIp will be output to the log
   */
  ip: boolean,
  /**
   * If this flag is enabled, $context.identity.caller will be output to the log
   */
  caller: boolean,
  /**
   * If this flag is enabled, $context.identity.user will be output to the log
   */
  user: boolean,
  /**
   * If this flag is enabled, $context.requestTime will be output to the log
   */
  requestTime: boolean,
  /**
   * If this flag is enabled, $context.httpMethod will be output to the log
   */
  httpMethod: boolean,
  /**
   * If this flag is enabled, $context.resourcePath will be output to the log
   */
  resourcePath: boolean,
  /**
   * If this flag is enabled, $context.status will be output to the log
   */
  status: boolean,
  /**
   * If this flag is enabled, $context.protocol will be output to the log
   */
  protocol: boolean,
  /**
   * If this flag is enabled, $context.responseLength will be output to the log
   */
  responseLength: boolean
}

/**
 * factory methods for access log format.
 */
export class AccessLogFormat {
  /**
   * Generate Common Log Format.
   */
  public static clf(): AccessLogOptions {
    return {
      format: `${AccessLogField.contextIdentitySourceIp()} ${AccessLogField.contextIdentityCaller()} ${AccessLogField.contextIdentityUser()} \
[${AccessLogField.contextRequestTime()}] "${AccessLogField.contextHttpMethod()} ${AccessLogField.contextResourcePath()} ${AccessLogField.contextProtocol()}" \
${AccessLogField.contextStatus()} ${AccessLogField.contextResponseLength()} ${AccessLogField.contextRequestId()}`
    };
  }

  /**
   * Access log will be produced in the JSON format with a set of fields most useful in the
   * access log. All fields are turned on by default with the option to turn off specific fields.
   */
  public static jsonWithStandardFields(
    fields: IJsonWithStandardFieldProps = {
      ip: true,
      user: true,
      caller: true,
      requestTime: true,
      httpMethod: true,
      resourcePath: true,
      status: true,
      protocol: true,
      responseLength: true
    }): AccessLogOptions {
    return {
      format: JSON.stringify({
        requestId: AccessLogField.contextRequestId(),
        ip: fields.ip ? AccessLogField.contextIdentitySourceIp() : undefined,
        user: fields.user ? AccessLogField.contextIdentityUser() : undefined,
        caller: fields.caller ? AccessLogField.contextIdentityCaller() : undefined,
        requestTime: fields.requestTime ? AccessLogField.contextRequestTime() : undefined,
        httpMethod: fields.httpMethod ? AccessLogField.contextHttpMethod() : undefined,
        resourcePath: fields.resourcePath ? AccessLogField.contextResourcePath() : undefined,
        status: fields.status ? AccessLogField.contextStatus() : undefined,
        protocol: fields.protocol ? AccessLogField.contextProtocol() : undefined,
        responseLength: fields.responseLength ? AccessLogField.contextResponseLength() : undefined
      })
    };
  }

  /**
   * Custom log format.
   * You can create any log format string. You can easily get the $ context variable by using the methods of AccessLogField.
   * @param format
   * @example custom(JSON.stringify({
   *  requestId: AccessLogField.contextRequestId(),
   *  sourceIp: AccessLogField.contextIdentitySourceIp(),
   *  method: AccessLogFiled.contextHttpMethod(),
   *  userContext: {
   *    sub: AccessLogField.contextAuthorizerClaims('sub'),
   *    email: AccessLogField.contextAuthorizerClaims('email')
   *  }
   * }))
   */
  public static custom(format: string): AccessLogOptions {
    return {format};
  }
}
