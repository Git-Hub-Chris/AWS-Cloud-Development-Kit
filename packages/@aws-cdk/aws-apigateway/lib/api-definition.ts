import * as s3 from '@aws-cdk/aws-s3';
import * as s3_assets from '@aws-cdk/aws-s3-assets';
import * as cdk from '@aws-cdk/core';
import { CfnRestApi } from './apigateway.generated';

/**
 * Represents an OpenAPI definition asset.
 */
export abstract class ApiDefinition {
  /**
   * Creates an API definition from a specification file in an S3 bucket
   */
  public static fromBucket(bucket: s3.IBucket, key: string, objectVersion?: string): S3ApiDefinition {
    return new S3ApiDefinition(bucket, key, objectVersion);
  }

  /**
   * Creates an API definition from a string
   */
  public static fromInline(definition: string): InlineApiDefinition {
    return new InlineApiDefinition(definition);
  }

  /**
   * Loads the API specification from a local disk asset.
   */
  public static fromAsset(file: string, options?: s3_assets.AssetOptions): AssetApiDefinition {
    return new AssetApiDefinition(file, options);
  }

  /**
   * Called when the specification is initialized to allow this object to bind
   * to the stack, add resources and have fun.
   *
   * @param scope The binding scope. Don't be smart about trying to down-cast or
   * assume it's initialized. You may just use it as a construct scope.
   */
  public abstract bind(scope: cdk.Construct): ApiDefinitionConfig;
}

/**
 * Post-Binding Configuration for a CDK construct
 */
export interface ApiDefinitionConfig {
  /**
   * The location of the specification in S3 (mutually exclusive with `inlineDefinition`).
   *
   * @default a new parameter will be created
   */
  readonly s3Location?: CfnRestApi.S3LocationProperty;

  /**
   * Inline specification (mutually exclusive with `s3Location`).
   *
   * @default a new parameter will be created
   */
  readonly inlineDefinition?: string;
}

/**
 * Swagger/OpenAPI specification from an S3 archive
 */
export class S3ApiDefinition extends ApiDefinition {
  private bucketName: string;

  constructor(bucket: s3.IBucket, private key: string, private objectVersion?: string) {
    super();

    if (!bucket.bucketName) {
      throw new Error('bucketName is undefined for the provided bucket');
    }

    this.bucketName = bucket.bucketName;
  }

  public bind(_scope: cdk.Construct): ApiDefinitionConfig {
    return {
      s3Location: {
        bucket: this.bucketName,
        key: this.key,
        version: this.objectVersion,
      },
    };
  }
}

/**
 * OpenAPI specification from an inline string (limited to 4KiB)
 */
export class InlineApiDefinition extends ApiDefinition {
  constructor(private definition: string) {
    super();

    if (definition.length === 0) {
      throw new Error('Inline API definition cannot be empty');
    }
  }

  public bind(_scope: cdk.Construct): ApiDefinitionConfig {
    return {
      inlineDefinition: this.definition,
    };
  }
}

/**
 * OpenAPI specification from a local file.
 */
export class AssetApiDefinition extends ApiDefinition {
  private asset?: s3_assets.Asset;

  constructor(private readonly path: string, private readonly options: s3_assets.AssetOptions = { }) {
    super();
  }

  public bind(scope: cdk.Construct): ApiDefinitionConfig {
    // If the same AssetAPIDefinition is used multiple times, retain only the first instantiation.
    if (this.asset === undefined) {
      this.asset = new s3_assets.Asset(scope, 'APIDefinition', {
        path: this.path,
        ...this.options,
      });
    }

    if (this.asset?.isZipArchive) {
      throw new Error(`Asset cannot be a .zip file or a directory (${this.path})`);
    }

    return {
      s3Location: {
        bucket: this.asset?.s3BucketName,
        key: this.asset?.s3ObjectKey,
      },
    };
  }
}