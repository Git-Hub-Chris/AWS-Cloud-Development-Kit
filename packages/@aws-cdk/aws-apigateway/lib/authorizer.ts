import { AuthorizationType } from './method';

/**
 * Represents an API Gateway authorizer.
 */
export interface IAuthorizer {
  /**
   * The authorizer ID.
   * @attribute
   */
  readonly authorizerId: string;

  /**
   * The required authorization type of this authorizer. If not specified,
   * `authorizationType` is required when the method is defined.
   */
  readonly authorizationType?: AuthorizationType;
}
