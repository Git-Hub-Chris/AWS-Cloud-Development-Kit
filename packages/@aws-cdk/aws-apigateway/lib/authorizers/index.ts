export * from './lambda';
export * from './authorizer-base';