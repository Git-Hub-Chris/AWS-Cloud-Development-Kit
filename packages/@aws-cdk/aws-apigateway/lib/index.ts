export * from './restapi';
export * from './restapi-ref';
export * from './resource';
export * from './method';
export * from './integration';
export * from './deployment';
export * from './stage';
export * from './integrations';
export * from './lambda-api';
export * from './api-key';
export * from './usage-plan';
export * from './usage-plan-key';

// AWS::ApiGateway CloudFormation Resources:
export * from './apigateway.generated';
