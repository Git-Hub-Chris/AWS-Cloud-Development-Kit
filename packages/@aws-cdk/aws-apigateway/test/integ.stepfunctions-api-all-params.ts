import * as sfn from '@aws-cdk/aws-stepfunctions';
import * as cdk from '@aws-cdk/core';
import { Construct } from 'constructs';
import * as apigw from '../lib';

/**
 * Stack verification steps:
 * * `curl -X POST 'https://<api-id>.execute-api.<region>.amazonaws.com/prod' \
 * * -d '{"key":"Hello"}' -H 'Content-Type: application/json'`
 * The above should return a "Hello" response
 */

class StepFunctionsRestApiDeploymentStack extends cdk.Stack {
  constructor(scope: Construct) {
    super(scope, 'StepFunctionsRestApiDeploymentStack');

    const passTask = new sfn.Pass(this, 'PassTask', {
      result: { value: 'Hello' },
    });

    const stateMachine = new sfn.StateMachine(this, 'StateMachine', {
      definition: passTask,
      stateMachineType: sfn.StateMachineType.EXPRESS,
    });

    const api = new apigw.StepFunctionsRestApi(this, 'StepFunctionsRestApi', {
      deploy: false,
      stateMachine: stateMachine,
      headers: true,
      querystring: true,
      path: true,
      requestContext: {
        accountId: true,
        apiId: true,
        apiKey: true,
        authorizerPrincipalId: true,
        caller: true,
        cognitoAuthenticationProvider: true,
        cognitoAuthenticationType: true,
        cognitoIdentityId: true,
        cognitoIdentityPoolId: true,
        httpMethod: true,
        stage: true,
        sourceIp: true,
        user: true,
        userAgent: true,
        userArn: true,
        requestId: true,
        resourceId: true,
        resourcePath: true,
      },
    });

    api.deploymentStage = new apigw.Stage(this, 'stage', {
      deployment: new apigw.Deployment(this, 'deployment', {
        api,
      }),
    });
  }
}

const app = new cdk.App();
new StepFunctionsRestApiDeploymentStack(app);
app.synth();
