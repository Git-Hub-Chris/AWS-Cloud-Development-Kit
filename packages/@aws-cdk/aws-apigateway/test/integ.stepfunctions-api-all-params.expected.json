{
  "Resources": {
    "StateMachineRoleB840431D": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": {
                  "Fn::Join": [
                    "",
                    [
                      "states.",
                      {
                        "Ref": "AWS::Region"
                      },
                      ".amazonaws.com"
                    ]
                  ]
                }
              }
            }
          ],
          "Version": "2012-10-17"
        }
      }
    },
    "StateMachine2E01A3A5": {
      "Type": "AWS::StepFunctions::StateMachine",
      "Properties": {
        "RoleArn": {
          "Fn::GetAtt": [
            "StateMachineRoleB840431D",
            "Arn"
          ]
        },
        "DefinitionString": "{\"StartAt\":\"PassTask\",\"States\":{\"PassTask\":{\"Type\":\"Pass\",\"Result\":\"Hello\",\"End\":true}}}",
        "StateMachineType": "EXPRESS"
      },
      "DependsOn": [
        "StateMachineRoleB840431D"
      ]
    },
    "StepFunctionsRestApiDeploymentStackStateMachineapiRoleNewD8D444E0": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        }
      }
    },
    "StepFunctionsRestApiDeploymentStackStateMachineapiRoleNewDefaultPolicy6DEF984E": {
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "states:StartSyncExecution",
              "Effect": "Allow",
              "Resource": {
                "Ref": "StateMachine2E01A3A5"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "PolicyName": "StepFunctionsRestApiDeploymentStackStateMachineapiRoleNewDefaultPolicy6DEF984E",
        "Roles": [
          {
            "Ref": "StepFunctionsRestApiDeploymentStackStateMachineapiRoleNewD8D444E0"
          }
        ]
      }
    },
    "StepFunctionsRestApiC6E3E883": {
      "Type": "AWS::ApiGateway::RestApi",
      "Properties": {
        "Name": "StepFunctionsRestApi"
      }
    },
    "StepFunctionsRestApiCloudWatchRoleB06ACDB9": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "apigateway.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs"
              ]
            ]
          }
        ]
      }
    },
    "StepFunctionsRestApiAccountBD0CCC0E": {
      "Type": "AWS::ApiGateway::Account",
      "Properties": {
        "CloudWatchRoleArn": {
          "Fn::GetAtt": [
            "StepFunctionsRestApiCloudWatchRoleB06ACDB9",
            "Arn"
          ]
        }
      },
      "DependsOn": [
        "StepFunctionsRestApiC6E3E883"
      ]
    },
    "StepFunctionsRestApiANY7699CA92": {
      "Type": "AWS::ApiGateway::Method",
      "Properties": {
        "HttpMethod": "ANY",
        "ResourceId": {
          "Fn::GetAtt": [
            "StepFunctionsRestApiC6E3E883",
            "RootResourceId"
          ]
        },
        "RestApiId": {
          "Ref": "StepFunctionsRestApiC6E3E883"
        },
        "AuthorizationType": "NONE",
        "Integration": {
          "Credentials": {
            "Fn::GetAtt": [
              "StepFunctionsRestApiDeploymentStackStateMachineapiRoleNewD8D444E0",
              "Arn"
            ]
          },
          "IntegrationHttpMethod": "POST",
          "IntegrationResponses": [
            {
              "ResponseTemplates": {
                "application/json": "#set($inputRoot = $input.path('$'))\n                #if($input.path('$.status').toString().equals(\"FAILED\"))\n                    #set($context.responseOverride.status = 500)\n                    { \n                      \"error\": \"$input.path('$.error')\",\n                      \"cause\": \"$input.path('$.cause')\"\n                    }\n                #else\n                    $input.path('$.output')\n                #end"
              },
              "StatusCode": "200"
            },
            {
              "ResponseTemplates": {
                "application/json": "{\n            \"error\": \"Bad input!\"\n          }"
              },
              "SelectionPattern": "4\\d{2}",
              "StatusCode": "400"
            },
            {
              "ResponseTemplates": {
                "application/json": "\"error\": $input.path('$.error')"
              },
              "SelectionPattern": "5\\d{2}",
              "StatusCode": "500"
            }
          ],
          "PassthroughBehavior": "NEVER",
          "RequestTemplates": {
            "application/json": {
              "Fn::Join": [
                "",
                [
                  "## Velocity Template used for API Gateway request mapping template\n##\n## This template forwards the request body, header, path, and querystring\n## to the execution input of the state machine.\n##\n## \"@@\" is used here as a placeholder for '\"' to avoid using escape characters.\n\n#set($inputString = '')\n#set($includeHeaders = true)\n#set($includeQueryString = true)\n#set($includePath = true)\n#set($allParams = $input.params())\n{\n    \"stateMachineArn\": \"",
                  {
                    "Ref": "StateMachine2E01A3A5"
                  },
                  "\",\n\n    #set($inputString = \"$inputString,@@body@@: $input.body\")\n\n    #if ($includeHeaders)\n        #set($inputString = \"$inputString, @@header@@:{\")\n        #foreach($paramName in $allParams.header.keySet())\n            #set($inputString = \"$inputString @@$paramName@@: @@$util.escapeJavaScript($allParams.header.get($paramName))@@\")\n            #if($foreach.hasNext)\n                #set($inputString = \"$inputString,\")\n            #end\n        #end\n        #set($inputString = \"$inputString }\")\n        \n    #end\n\n    #if ($includeQueryString)\n        \n        #set($inputString = \"$inputString, @@querystring@@:{\")\n        #foreach($paramName in $allParams.querystring.keySet())\n            #set($inputString = \"$inputString @@$paramName@@: @@$util.escapeJavaScript($allParams.querystring.get($paramName))@@\")\n            #if($foreach.hasNext)\n                #set($inputString = \"$inputString,\")\n            #end\n        #end\n        #set($inputString = \"$inputString }\")\n    #end\n\n    #if ($includePath)\n        #set($inputString = \"$inputString, @@path@@: @@$context.resourcePath@@\")\n    #end\n    \n    #set($requestContext = \"@@requestContext@@: {@@accountId@@:@@$context.identity.accountId@@,@@apiId@@:@@$context.apiId@@,@@apiKey@@:@@$context.identity.apiKey@@,@@authorizerPrincipalId@@:@@$context.authorizer.principalId@@,@@caller@@:@@$context.identity.caller@@,@@cognitoAuthenticationProvider@@:@@$context.identity.cognitoAuthenticationProvider@@,@@cognitoAuthenticationType@@:@@$context.identity.cognitoAuthenticationType@@,@@cognitoIdentityId@@:@@$context.identity.cognitoIdentityId@@,@@cognitoIdentityPoolId@@:@@$context.identity.cognitoIdentityPoolId@@,@@httpMethod@@:@@$context.httpMethod@@,@@stage@@:@@$context.stage@@,@@sourceIp@@:@@$context.identity.sourceIp@@,@@user@@:@@$context.identity.user@@,@@userAgent@@:@@$context.identity.userAgent@@,@@userArn@@:@@$context.identity.userArn@@,@@requestId@@:@@$context.requestId@@,@@resourceId@@:@@$context.resourceId@@,@@resourcePath@@:@@$context.resourcePath@@}\")\n    ## Check if the request context should be included as part of the execution input\n    #if($requestContext && !$requestContext.empty)\n        #set($inputString = \"$inputString,\")\n        #set($inputString = \"$inputString $requestContext\")\n    #end\n\n    #set($inputString = \"$inputString}\")\n    #set($inputString = $inputString.replaceAll(\"@@\",'\"'))\n    #set($len = $inputString.length() - 1)\n    \"input\": \"{$util.escapeJavaScript($inputString.substring(1,$len))}\"\n}\n"
                ]
              ]
            }
          },
          "Type": "AWS",
          "Uri": {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":apigateway:",
                {
                  "Ref": "AWS::Region"
                },
                ":states:action/StartSyncExecution"
              ]
            ]
          }
        },
        "MethodResponses": [
          {
            "ResponseModels": {
              "application/json": "Empty"
            },
            "StatusCode": "200"
          },
          {
            "ResponseModels": {
              "application/json": "Error"
            },
            "StatusCode": "400"
          },
          {
            "ResponseModels": {
              "application/json": "Error"
            },
            "StatusCode": "500"
          }
        ]
      }
    },
    "deployment33381975b5dafda9a97138f301ea25da405640e8": {
      "Type": "AWS::ApiGateway::Deployment",
      "Properties": {
        "RestApiId": {
          "Ref": "StepFunctionsRestApiC6E3E883"
        }
      },
      "DependsOn": [
        "StepFunctionsRestApiANY7699CA92"
      ]
    },
    "stage0661E4AC": {
      "Type": "AWS::ApiGateway::Stage",
      "Properties": {
        "RestApiId": {
          "Ref": "StepFunctionsRestApiC6E3E883"
        },
        "DeploymentId": {
          "Ref": "deployment33381975b5dafda9a97138f301ea25da405640e8"
        },
        "StageName": "prod"
      }
    }
  }
}